core/convenience.js

```diff
@@ -64,8 +64,8 @@
     rp.key = password;
 
     /* do the encryption */
-    if (p.mode === "ccm" && sjcl.arrayBuffer && sjcl.arrayBuffer.ccm && plaintext instanceof ArrayBuffer) {
-      p.ct = sjcl.arrayBuffer.ccm.encrypt(prp, plaintext, p.iv, adata, p.ts);
+    if (p.mode === "ccm" && sjcl.arrayBuffer && sjcl.arrayBuffer.ccm) {
+      p.ct = sjcl.arrayBuffer.ccm.compat_encrypt(prp, plaintext, p.iv, adata, p.ts);
```

Comment:
Padding of ArrayBuffer is needed otherwise we get:

        mac[0] ^= data.getUint32(i);
                       ^
	Error: Index out of range.
    		at Object.sjcl.arrayBuffer.ccm._computeTag (evalmachine.<anonymous>:4952:24)
    		at Object.sjcl.arrayBuffer.ccm.encrypt (evalmachine.<anonymous>:4865:32)

In the first version padding was applied in compat_encrypt, however the encryption failed when encrypt was used directly (but there was no test for it).
 
To summarize what was now done:
 - CCM ArrayBuffer encryption now accepts only ArrayBuffer input
 - I removed compat_encrypt and compat_decrypt which were meant for encryption/decryption when input is a bitArray - now this is not needed as sjcl.codec.arrayBuffer.fromBits was fixed (removed padBuffer at the end - padBuffer is now called directly in encrypt and decrypt functions) and bitArray can be easily encrypted using CCM ArrayBuffer encryption simply by converting into ArrayBuffer prior the encryption (see tests):
	var plaintextArrayBuffer = sjcl.codec.arrayBuffer.fromBits(plaintextBits);
  	var ciphertext = sjcl.encrypt(sessionKey, plaintextArrayBuffer, params);
 - encode and decode in convenience.js have been modified to encode/decode ArrayBuffer into Base64, for this reason fromBase64 and toBase64 have been added to sjcl.codec.arrayBuffer (in the previous version the ArrayBuffer was converted into bitArray before encoding because encode didn't support ArrayBuffers)
 - related to the previous point: encoding of ArrayBuffer provides two properties (ciphertextBuffer and ciphertextTag) instead of only one (ct) - this way decoding knows whether to decode into ArrayBuffer or bitArray (sjcl.decrypt output is thus the same as sjcl.encrypt input)